### PR TITLE
Controller support, hat collectable, and fixed level transition.

### PR DESCRIPTION
**Pull request by AidanCotsakis**
## Description:
- Added controller support: This was done mostly using unity's UI manager (Tested on Ps4 and Ps5 controllers)
X - Jump
Square - Boost
Left Joystick - Rotate
Options - Pause/Unpause
X - Restart
Circle - Main Menu
- Added hat collectable: Used to save the player once from hitting the ground, applied a small force to help the player recover, will be consumed upon use, does not transfer between levels.
- Cleaned up level transition: At the start of each level the level designer can choose a fixed starting velocity instead of saving and applying the previous velocity. At the start of the level the level designer can choose a minimum x position of the camera to allow the player to slide into view when the level is loaded.
- Tweaked game physics (This still needs some work)

## Images/Videos:

https://github.com/dominiccupidon/ubc-toboggan-game/assets/47440253/d9707f79-329b-49cc-bad2-b18a5163b0b7

## Checklist:
- [ X ] Title
- [ X ] Description of work
- [ X ] Images or videos
- [ X ] Notified team of PR

### CODE REVIEW

File: src/UBC Toboggan/Assets/Scripts/cameraFollow.cs [hunk 1]
```diff
@@ -7,19 +7,20 @@ public class cameraFollow : MonoBehaviour
...
     public float yOffset = 1f;
     public float xOffset = 1f;
+    public float minX = 1f;
 
     public Transform target;
 
     void Start()
     {
-        Vector3 newPos = new Vector3(target.position.x + xOffset, target.position.y + yOffset,-10f);
+        Vector3 newPos = new Vector3(Mathf.Max(target.position.x + xOffset, minX), target.position.y + yOffset,-10f);
```
Comment (dominiccupidon) [inline src/UBC Toboggan/Assets/Scripts/cameraFollow.cs:16]: Why did you change the new position formula?

File: src/UBC Toboggan/Assets/Scripts/PlayerControls.cs
Comment (dominiccupidon) [inline src/UBC Toboggan/Assets/Scripts/PlayerControls.cs:1]: Is it possible for you to explain what this file does?

File: src/UBC Toboggan/Assets/Scripts/cameraFollow.cs (hunk 1 above)
Comment (AidanCotsakis) [reply src/UBC Toboggan/Assets/Scripts/cameraFollow.cs:16]: This is to snap the camera to the starting position, I changed it to add the ability to have a minimum X value the camera can go, this allows the player to slide into screen rather than starting with the player in the "middle of the screen"

Described here probably a little better: "Cleaned up level transition: At the start of the level the level designer can choose a minimum x position of the camera to allow the player to slide into view when the level is loaded."

**Review by chandhana9 — APPROVED** [review]

**Review by dominiccupidon — APPROVED** [review]
